TrustModel.md

```diff
@@ -0,0 +1,426 @@
+ - Preparation and build of the trusted boot image for the TEE. 
+ - Signatures and encryption keys for containers are prepared and available
+ 
+````` Completing building the boot image triggers the next phase.`````
+
+#### Distribute
+The Distribute stage includes all preparations before the guest executes its first instructions, 
+and it will run some general and TEE-specific operations.
+- configuring the cluster with TEE
+- making the boot image available
+- taking initial measurements
+- provision secrets.
+- allocation of guest resource
+- loading guest fw and configuration occurs 
+- TEE-specific steps relating to preparing attestation/secrets etc
+
+The involved components include the ones that directly associated with the
+protections that TEEs provide (e.g, tdx seam-loader, tdx module, psp/sev firmware and so on),
+Pre-OS (e.g, guest firmware, guest kernel, initrd and so on) and specific components (e.g,
+kata-agent, attestation-agent and so on). Also includes preparing secrets for delivery to the TEE.
+
+```The first instruction being executed within the trusted boot image happens in the next phase```
+
+#### Deploy
+  - pod sandbox preparation
+  - "pre flight checks" include attestation measurment checks
+  - secret provisioning linked to boot attestation
+  
+  ```Receiving the request to pull and deploy a container happens in the runtime phase```
+
+
+#### Runtime
+- post boot attestation and secret provisioning
+- procedure of image pulling, unpacking and 
+  launching.
+- the workload in the container starts to  run in TEEs.
+
+
+These execution phases are sufficient to describe the execution of any forms of confidential
+container. 
+
+---
+---
+
+## Goals and not-goals
+
+### Goals
+
+The commonality of goals between confidential container and confidential computing are to reduce
+the ability for the platform owner, operator and attacker to access data and code inside TEEs
+sufficiently such that this path is not an economically or logically viable attack during
+execution[^1]  5.1 Goal.
+
+Around the three execution stages of confidential container, the scope of the threat model of
+confidential container includes:
+
+<* ADD diagram*>
+
+- Pre-container stage
+  - Measure the TEE where confidential containers run, and store the measurements to a secure location.
+
+- Container launch stage
+  - Measure and attest the TEE where confidential containers run to prove the deployment and initialization of pre-container are valid and expected;
+  - Pull the encrypted and signed container image in TEE;
+  - Provision the runtime configurations through a secure and attested control plane with the protections of confidentiality and integrity.
+
+- Workload operation stage
+  - Provide the protections of confidentiality and integrity for the storage and transport of data derived from TEE outside the TEE instance;
+  - Keep the dynamic measurement of TEE where confidential containers run, and store the measurements to a secure location;
```

Comment:
What do you mean by dynamic measurement? Typically TEEs have a static measurement taken at launch time. You can add software measurements of things inside the TEE, but they aren't of the TEE itself. I'm happy to be educated on something I've missed though.

Reply:
I think TDX has some form of dynamic measurement.

Reply:
TDX has runtime measurements. Kinda depends on what one means by dynamic. Here runtime means that you can measure stuff after boot/launch.

Reply:
> TDX has runtime measurements. Kinda depends on what one means by dynamic. Here runtime means that you can measure stuff after boot/launch.

Yes it is. SGX, SEV(-ES), SEV-SNP and TDX can support runtime measurements to report an attestation report in runtime, regardless of the capabilities, such as including user-defined random report data and hash-extended dynamic measurement registries.

Moreover, only TDX can support the so-called dynamic measurements in hardware, such as RTMRs, or vTPM software solution. Additionally, SGX and SEV-SNP and TDX can support to include user-defined random report data in attestation report.

There are so many dimensions to evaluate the capability of measurements, so using dynamic measurement to describe the use case.